tests/field/example-django.sh

```diff
@@ -119,21 +119,46 @@ git add -v .
 git commit -m 'Add friendly welcome page'
 git push -u origin feature/welcome-page --force
 
-log 4 'Create merge request'
+log 4 'Add an untested feature'
+sed -E \
+    -e "s|^(from hello import views)$|\1\n\n\ndef trigger_error(request):\n\
+    \"\"\"Generate an error to demonstrate Sentry\"\"\"\n\
+    division_by_zero = 1 / 0\n\
+    return division_by_zero\n|" \
+    -i application/urls.py
+
+sed -E \
+    -e "s|^(urlpatterns = \[)$|\1\n    path('sentry-debug/', trigger_error),|" \
+    -i application/urls.py
+
+git add -v .
+git commit -m 'Add a very special additional feature'
+git push -u origin feature/welcome-page --force
+
+log 5 'Create merge request'
 gitlab POST merge_requests \
     --form "source_branch=feature/welcome-page" \
     --form "target_branch=master" \
     --form "title=Add friendly welcome page" \
     --form "description=A minimal Django application that shows some text. Tests are included." \
     > /dev/null
 
-log 5 'Allow pipeline to build and push an image'
+log 6 'Allow pipeline to build and push an image'
 for minutes in $(seq 13 -1 1); do
     echo "- Waiting... ($minutes' remaining)"
     sleep 1m
 done
 
-log 6 'Trigger production relase'
+log 7 'Trigger production relase'
 git checkout master
 git tag 1.0.0
 git push --tags --force
+
+log 8 'Allow pipeline to deploy an image'
+for minutes in $(seq 3 -1 1); do
+    echo "- Waiting... ($minutes' remaining)"
+    sleep 1m
+done
+
+log 9 'Trigger an error incident'
+curl -X GET https://application-route-development-example-django.appuioapp.ch/sentry-debug/
```

Comment:
We could save the 3-minutes wait time, and having to code the specifics of the development route in this field test, if we were to trigger the "buggy code" (by help of a management command) via a cron job.

Reply:
Yeah, it would be cleaner to trigger it from a cron job.  As we discussed, I did this implementation before you suggested using a cron job.  I can go ahead and make that update.

Reply:
Actually, on second thought, I'm not sure that adding the error through a custom management command is the best way to go.  I've tested, and Sentry does register the error, but I have a few objections:

- It doesn't really illustrate the type of error that we're installing sentry for. It's for investigating errors generated by accessing the website, not errors generated by admin users calling commands inside the container.
- It's nice for demo purposes to be able to generate an incident at will just from a browser.  Sure we could include both error paths, but then it starts getting more complex and cluttered, less elegant.
- Generating the error from the cron job rather than from the field test script directly would probably be cleaner, but if we go with that option, I'd prefer to generate it with a curl call from the container rather than from manage.py on the command-line for the reasons I listed above.

Reply:
I agree. Being able to trigger issues on demand is actually helpful in demos (aka talks). :+1: